xla/pjrt/gpu/se_gpu_pjrt_client.cc

```diff
@@ -1248,6 +1249,15 @@ absl::StatusOr<DeviceTopologyPair> BuildDistributedDevices(
   }
   gpu_executable_run_options->set_gpu_global_device_ids(
       std::move(gpu_device_ids));
+#if GOOGLE_CUDA
+  static const bool xla_gpu_experimental_enable_nvshmem =
```

Comment:
This should not be here:
1. No new GOOGLE_CUDA when it can be avoided
2. Do not make nvshmem public and include it in SE (or anywhere outside for gpu/collectives folder)

This should go under GpuCollectives API and be implemented in NcclCollectives and NvsmemCollectives. The whole point of this API is so that we don't leak nvshmem/nccl/anything-else all over XLA:GPU codebase.

Reply:
This particular code is specific to nvshmem since it requires a global initialization, so there wouldn't be any equivalent abstraction on the nccl side.
However, I can hide it behind something like:
`CollectivesRegistry::Get("gpu", "nvshmem").SetEnvInfo(...)`
Would that work?